superset-frontend/src/chart/ChartRenderer.jsx

```diff
@@ -222,13 +222,21 @@ class ChartRenderer extends React.Component {
       queryResponse,
     } = this.props;
 
+    let chartClassName = snakeCase(vizType);
+    // It's bad practice to use unprefixed `vizType` as classnames for chart
+    // container. It may cause css conflicts as in the case of table chart.
+    // When migrating legacy chart types, we should gradually add the prefix
+    // `superset-chart-` to each one of them.
+    chartClassName =
```

Comment:
```ts
const snakeCaseVizType = snakeCase(vizType);
const chartClassName = vizType === 'table' ? `superset-chart-${snakeCaseVizType}` : snakeCaseVizType;
```

Reply:
I generally avoid reassigning variables. `NVD3Vis.js` is an example of many reassignments that made it very difficult to refactor because order of operations matter a lot.

Reply:
> What do you think of
> 
> ```ts
> const snakeCaseVizType = snakeCase(vizType);
> const chartClassName = vizType === 'table' ? `superset-chart-${snakeCaseVizType}` : snakeCaseVizType;
> ```

I like this one.

Reply:
> I generally avoid reassigning variables. `NVD3Vis.js` is an example of many reassignments that made it very difficult to refactor because order of operations matter a lot.

I think it makes sense to use a combination of both. `const` should be default, but `let` can be acceptable in small scopes.